### ISSUE TITLE
RethinkDB Crashed

### ISSUE THREAD

**Issue by alehano** [rethinkdb/rethinkdb#6570]
One day, RethinkDB crashed. Here is the log. 

Exiting.
Posted by rethinkdb1

about an hour ago
Wed Dec 20 07:31:03 2017
```
1 [0x7d6e20]: backtrace_t::backtrace_t() at 0x7d6e20 (rethinkdb)
2 [0x7d7199]: format_backtrace(bool) at 0x7d7199 (rethinkdb)
3 [0x72b053]: report_fatal_error(char const*, int, char const*, ...) at 0x72b053 (rethinkdb)
4 [0x7e35ce]: extproc_spawner_t::spawn(object_buffer_t<socket_stream_t>*, int*) at 0x7e35ce (rethinkdb)
5 [0x7e0148]: extproc_worker_t::acquired(signal_t*) at 0x7e0148 (rethinkdb)
6 [0x7df413]: extproc_job_t::extproc_job_t(extproc_pool_t*, bool (*)(read_stream_t*, write_stream_t*), signal_t*) at 0x7df413 (rethinkdb)
7 [0x7d8992]: http_runner_t::http(http_opts_t const&, http_result_t*, signal_t*) at 0x7d8992 (rethinkdb)
8 [0x97ec9c]: ql::dispatch_http(ql::env_t*, http_opts_t const&, http_runner_t*, http_result_t*, ql::bt_rcheckable_t const*) at 0x97ec9c (rethinkdb)
9 [0xc41e4d]: version_checker_t::do_check(bool, auto_drainer_t::lock_t) at 0xc41e4d (rethinkdb)
10 [0xc43144]: callable_action_instance_t<std::_Bind<std::_Mem_fn<void (version_checker_t::*)(bool, auto_drainer_t::lock_t)> (version_checker_t*, bool, auto_drainer_t::lock_t)> >::run_action() at 0xc43144 (rethinkdb)
11 [0x815857]: coro_t::run() at 0x815857 (rethinkdb)
```
Posted by rethinkdb1

about an hour ago
Backtrace:
Posted by rethinkdb1

about an hour ago
Guarantee failed: [res == 0] (errno 32 - Broken pipe) could not send socket file descriptor to worker process
Posted by rethinkdb1

about an hour ago
Error in src/extproc/extproc_spawner.cc at line 221:
Posted by rethinkdb1

(Edited by @srh: replaced \n's in backtrace with newlines.)

**Comment by srh** [comment]
Thanks for the report.  What OS and platform were you running?

It looks like the best course of action would be to accept that the OS might sometimes refuse to let us send file descriptors to worker processes for some reason, and swallow the error or return it back to the user.  We already have the infrastructure for that.

In this case the caller is the update checker.  This particular crash wouldn't happen if you pass `--no-update-check`.  I recommend that option heartily for any user -- personally I'd remove the update checker entirely.

**Comment by alehano** [comment]
It runs in Docker under Ubuntu. 
Thanks for an advice.

**Comment by srh** [comment]
Closing as a dupe of #5572.

**Comment by srh** [comment]
Fixed with PR #6572.  Thanks for bringing this to my attention.
